src/vsr/replica.zig

```diff
@@ -1035,6 +1093,18 @@ pub fn ReplicaType(
                 prepare.message.header.checksum,
             });
 
+            assert(self.prepare_timeout.ticking);
+            assert(self.primary_abdicate_timeout.ticking);
+            assert(!self.primary_abdicating);
+            if (self.primary_pipeline_pending() == null) {
+                self.prepare_timeout.stop();
+                self.primary_abdicate_timeout.stop();
+            } else {
+                if (prepare == self.pipeline.queue.prepare_queue.head_ptr().?) {
```

Comment:
Hm, I was thinking more about `self.primary_pipeline_pending().?` rather than literally the first one (the headone might be waitin for disk, and the secodn might be waiting for quorum)

Reply:
Though, I think this should also be fine.